docs/stress-test.md

```diff
@@ -0,0 +1,124 @@
+One way to avoid this is run the script with a `--count` that is not higher than 500 which
+gives time between runs for the auto-scaler to add more `user` nodes.
+
+As an example, on IBM Cloud there is a hard [pod limit](https://cloud.ibm.com/docs/containers?topic=containers-limitations)
+of 110 per node and there are about 25 system pods per node. Our testing cluster user notebooks are using a micro
+profile so their resource usage is not an issue, they are just limited to the 110 pod-per-node limit.
+As a reference, to scale up to 3000 users/pods we need to have at least 35 user nodes.
+
+### Steady state testing
+
+The `--keep` option can be used to scale up the number of pods in the cluser and retain them
+so that you can perform tests or profiling on the hub with a high load. When the script runs
+it will first check for the number of existing `hub-stress-test` users and start creating new
+users based on an index so you can run the script with a `--count` value of 200-500 if you need
+to let the auto-scaler add `user` nodes after each run.
+
+Note that the `c.NotebookApp.shutdown_no_activity_timeout` value in the user notebook image (in the
+testing cluster) should either be left at the default (0) or set to some larger window so that while
+you are scaling up the notebook pods do not shut themselves down.
+
+## Scaling down
+
+If you used the `--keep` option to scale up and retain pods for steady state testing, when you are
+done you can scale down the pods and users by using the `--purge` option. The users created by the
+script all have a specific naming convention so it knows which notebook servers to stop and users
+to remove.
+
+## Monitoring
+
+Depending on the number of pods being created or deleted the script can take awhile. During a run
+there are some dashboards you should be watching and also the hub logs. The logging and monitoring
+platform is deployment-specific but the following are some examples of dashboards we monitor:
+
+* `Jupyter Notebook Health (Testing)`
+  This dashboard shows the active user notebook pods, nodes in the cluster and `user-placeholder`
+  pods. This is mostly interesting to watch the active user notebook pods go up or down when scaling
+  up or down with the script. The placeholders and user nodes may also fluctuate as placeholder pods
+  are pre-empted and as the auto-scaler is adding or removing user nodes.
+
+  ![hub-stress-test-health](images/hub-stress-test-health.png)
+
+* `Jupyter Hub Golden Signals (testing)`
+  This is where you can monitor the response time and request rate on the hub. As user notebook pods
+  are scaled up each of those pods will "check in" with the hub to report their activity. By default
+  each pod checks in with the hub every [5 minutes](https://github.com/jupyterhub/jupyterhub/blob/5dee864af/jupyterhub/singleuser.py#L463). So we expect that the more active user notebook pods in the cluster will increase
```

Comment:
Since this doc was written we've changed `JUPYTERHUB_ACTIVITY_INTERVAL` to be 3600 (1 hour). It would be good to have a section in the docs on performance tweaks we've made to our hub, but that could be a follow up.

Reply:
Opened #3 for tracking this.